vyper/semantics/analysis/module.py

```diff
@@ -565,7 +566,15 @@ def visit_ExportsDecl(self, node):
 
                 self._add_exposed_function(func_t, item, relax=False)
                 with tag_exceptions(item):  # tag exceptions with specific item
-                    self._self_t.typ.add_member(func_t.name, func_t)
+                    try:
+                        self._self_t.typ.add_member(func_t.name, func_t)
+                    except NamespaceCollision as e:
+                        export_name = item.node_source_code
+                        # Re-raise with more specific message mentioning the export
+                        raise NamespaceCollision(
+                            f"Member '{func_t.name}' already exists in self (when exporting `{export_name}`)",
+                            prev_decl=e.prev_decl
+                        ) from e
```

Comment:
this try/catch block seems a little ad hoc to me. would it be better to fix it at the source (where NamespaceCollision is raised in the first place)?